code-scanning/eslint.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: ESLint
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+jobs:
+  eslint:
+    name: Run eslint scanning
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```

Comment:
should we start using v3? I saw it was released a few days ago

Reply:
The checkout v3 readme says:
> - Updated to the node16 runtime by default
>   - This requires a minimum [Actions Runner](https://github.com/actions/runner/releases/tag/v2.285.0) version of v2.285.0 to run, which is by default available in GHES 3.4 or later.

Any possibility the server has an old version runner and cannot run v3 checkout?

Reply:
Please use v3 version. New server releases have new runner and hence won't be a problem.

Reply:
@yongyan-gh , you missed this one

Reply:
updated version to v3